### PR TITLE
Add jenkinsfile

### PR DESCRIPTION
**Pull request by MatMoore**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env groovy
+
+REPOSITORY = 'govuk_taxonomy_helpers'
```
Comment (suzannehamilton) [inline Jenkinsfile]: Unused constant?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      checkout scm
+    }
+
+    stage('Clean') {
+      govuk.cleanupGit()
+      govuk.mergeMasterBranch()
+    }
+
+    stage('Bundle') {
+      govuk.bundleApp()
```
Comment (suzannehamilton) [inline Jenkinsfile]: We found on another gem that we couldn't use `bundleApp`. I think the problem was that it adds the `--deployment` flag, which isn't appropriate for a gem.

We used `sh("bundle install --path ${JENKINS_HOME}/bundles/${JOB_NAME}")` instead but we should probably extract it to the Jenkins helpers in govuk-puppet.

**Review by suzannehamilton — CHANGES_REQUESTED** [review]
This all looks sensible to me, but I suspect you'll have to update the `bundleApp` step to get the build working on Jenkins.

File: Jenkinsfile (hunk 1 above)
Comment (MatMoore) [reply Jenkinsfile]: I'm not sure actually, I can't find where this variable is used, but it's in the rails template Jenkinsfile, and gds-api-adapters (which I looked at as an example of another gem) also has it. 😕

File: Jenkinsfile (hunk 2 above)
Comment (MatMoore) [reply Jenkinsfile]: Ah thanks, I'll look into extracting that then. Do you know if it's sufficient to just have `bundleGem` and `bundleApp` or do we use bundler in other ways?

File: Jenkinsfile (hunk 1 above)
Comment (suzannehamilton) [reply Jenkinsfile]: It's not in the [A/B testing gem Jenkinsfile](https://github.com/alphagov/govuk_ab_testing/blob/master/Jenkinsfile), and the rails template uses the constant in the integration deployment step. I'm not aware of any Jenkins magic that needs a `REPOSITORY` constant - I think it's safe to just delete.

File: Jenkinsfile (hunk 2 above)
Comment (suzannehamilton) [reply Jenkinsfile]: I haven't seen any other uses of it, and it looks like [we're not using it in any other ways at the moment](https://github.com/search?l=Groovy&p=4&q=org%3Aalphagov+%22bundle+install%22&type=Code).

**Comment by MatMoore** [comment]
@suzannehamilton Have extracted the bundle step in https://github.com/alphagov/govuk-puppet/pull/5502

**Pull request merged**